sycl/source/detail/memory_manager.hpp

```diff
@@ -172,14 +172,14 @@ class __SYCL_EXPORT MemoryManager {
                        sycl::detail::pi::PiEvent *OutEvent);
 
   static void fill_usm(void *DstMem, QueueImplPtr Queue, size_t Len,
-                       int Pattern,
+                       const std::vector<char> &Pattern,
```

Comment:
It is a little silly, but this is technically an ABI break because `MemoryManager` is exported despite not being used in or available to our headers.

Solution here would be to make a new overload of `fill_usm`, steal the implementation and redirect the old implementation, a little like you see in the one below. Please add a "TODO" to it too, similar to the other one.

Reply:
Yeah I wasn't sure but I suspected this PR would cause an ABI break, thanks for the suggestion, I'll add the overload